libraries/AP_HAL_ChibiOS/Scheduler.cpp

```diff
@@ -95,6 +95,14 @@ THD_WORKING_AREA(_monitor_thread_wa, MONITOR_THD_WA_SIZE);
 #define AP_HAL_CHIBIOS_IN_EXPECTED_DELAY_WHEN_NOT_INITIALISED 1
 #endif
 
+#if defined(STM32H7) && defined(IOMCU_FW)
```

Comment:
```suggestion
#if AP_REBOOT_ON_MEMGUARD_ERROR
```
... and default it off and define to true in the hwdef.

Reply:
I think that can be a bit deceptive If I do it the way you mentioned. 

As the normal AP_memory_guard_error is defined inside AP_InternalError. This is actually a special case for IOMCU_FW running  on STM32H7.